eclipse-collections-code-generator/src/main/resources/api/primitiveIterable.stg

```diff
@@ -127,6 +127,96 @@ public interface <name>Iterable extends PrimitiveIterable
         }
     }
 
+    /**
+     * Returns true if any of the values specified in the source array are contained
+     * in the <name>Iterable, and false if they are not.
+     *
+     * @since 11.0
+     */
+    default boolean containsAny(<type>... source)
+    {
+        <name>Iterable inside = this;
+        if (this.size() > 32 && source.length > 32 && !(this instanceof <name>Set))
+        {
+            inside = this.toSet();
+        }
+        for (<type> item : source)
+        {
+            if (inside.contains(item))
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Returns true if any of the values specified in the source <name>Iterable are contained
+     * in the <name>Iterable, and false if they are not.
+     *
+     * @since 11.0
+     */
+    default boolean containsAny(<name>Iterable source)
+    {
+        <name>Iterable outside = this;
+        <name>Iterable inside = source;
+        if (this.size() \< source.size())
+        {
+            outside = source;
+            inside = this;
+        }
+        if (inside.size() > 32 && !(inside instanceof <name>Set))
+        {
+            inside = inside.toSet();
+        }
```

Comment:
I've been thinking about two special cases in the `containsAny/None` situation:
case 1:
this: large list, say 1M entries
other: large set, larger than this, say 2 M entries
Our logic will create a 1M entry set to do the operation. This is not necessary. 
case 2:
this: large set, say 1M entries
other: large set, larger than this, say 2 M entries
We won't create a set in this case, but the loop order is wrong: this should be on the outside.

Both can be fixed by changing this code to:
```java
        if (outside instance of <name>Set)
        {
            <name>Iterable tmp = outside;
            outside = inside;
            inside = tmp;
        }
        else if (inside.size() > 32 && !(inside instanceof <name>Set))
        {
            inside = inside.toSet();
        }
```

Reply:
Hi @rinatigati, any questions or updates on the suggestion from @mohrezaei above?

Reply:
@donraab Nope! Sounded good to me. I updated the PR with @mohrezaei's suggestions.

Looks like there were artifact transfer failures/connection problems with a couple of the checks. Will have to re-run those somehow.